File: Tests/Oracle/corpus/sample.go
package demo

type Point struct {
	X int
	Y int
}

type Shape interface {
	Area() float64
}

func Total(values []int) int {
	sum := 0
	for _, v := range values {
		sum += v
	}
	switch {
	case sum > 10:
		return 10
	default:
		return sum
	}
}

File: Tests/Oracle/corpus/sample.rs
mod demo {
    pub struct Point {
        pub x: i32,
        pub y: i32,
    }

    pub enum Shape {
        Circle,
        Square,
    }

    pub fn total(values: &[i32]) -> i32 {
        let mut sum = 0;
        for v in values {
            sum += v;
        }
        match sum {
            0 => 0,
            n => n,
        }
    }
}
